### PR TITLE
chore: surface changelogs to the doc site

### PR DESCRIPTION
**Pull request by TarunAdobe**
## Description

This PR surfaces the changelogs for every component to the doc site so that consumers can check what breaking changes or features were added in which release and plan their updates in a better way.

## How has this been tested?

-   [ ] _Test case 1_
    1. Go here
    2. Do this
-   [ ] _Test case 2_
    1. Go here
    2. Do this

## Screenshots (if appropriate)

image

## Types of changes

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to change)
-   [x] Chore (minor updates related to the tooling or maintenance of the repository, does not impact compiled assets)

## Checklist

-   [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
-   [x] My code follows the code style of this project.
-   [x] If my change required a change to the documentation, I have updated the documentation in this pull request.
-   [x] I have read the **[CONTRIBUTING](<(https://github.com/adobe/spectrum-web-components/blob/main/CONTRIBUTING.md)>)** document.
-   [x] I have added tests to cover my changes.
-   [x] All new and existing tests passed.
-   [x] I have reviewed at the Accessibility Practices for this feature, see: [Aria Practices](https://www.w3.org/TR/wai-aria-practices/)

## Best practices

This repository uses conventional commit syntax for each commit message; note that the GitHub UI does not use this by default so be cautious when accepting suggested changes. Avoid the "Update branch" button on the pull request and opt instead for rebasing your branch against `main`.

### CODE REVIEW

**Comment by Rajdeepc** [comment]
Great Start! Thanks for doing this!
Found few formatting issues though which seems to be trivial but let's fix this up!
Screenshot 2024-07-08 at 3 02 58 PM

Also let's make sure we communicate the users about no changelogs as seen in `sp-accordion-item`

Screenshot 2024-07-08 at 3 02 45 PM

File: projects/documentation/scripts/copy-component-docs.js [hunk 1]
```diff
@@ -233,6 +300,7 @@ async function main() {
         `${projectDir}/(packages|tools)/**/*.md`
     )) {
         await processREADME(mdPath);
+        await processChangelog(mdPath);
```
Comment (Rajdeepc) [inline projects/documentation/scripts/copy-component-docs.js:303]: Why this needs to be synchronous? Do we need to wait for processREADME to complete?

**Review by Rajdeepc — CHANGES_REQUESTED** [review]

File: projects/documentation/src/router.ts [hunk 2]
```diff
@@ -18,16 +18,25 @@ tabs.addEventListener('change', (event: Event) => {
     const { selected } = target;
     const { pathname } = location;
     const isAPI = pathname.search('api') > -1;
+    const isChangelog = pathname.search('changelog') > -1;
     switch (selected) {
         case 'api': {
             if (isAPI) return;
-            const dest = (pathname + '/api/').replace('//a', '/a');
+            const dest =
+                pathname.replace(/\/api\/?|\/changelog\/?/, '/') + 'api/';
```
[the remark this answers is not in the document]
Comment (TarunAdobe) [reply projects/documentation/src/router.ts]: yessss sure.

File: projects/documentation/scripts/copy-component-docs.js (hunk 1 above)
Comment (TarunAdobe) [reply projects/documentation/scripts/copy-component-docs.js]: no we don't. i've made it synchronous now.

**Review by Rajdeepc — APPROVED** [review]
Thanks for bringing this in!

**Pull request merged**